### PR TITLE
dpms: add button_off, button_toggle, cache_timeout

### PR DESCRIPTION
**Pull request by lasers**
This adds missing `cache_timeout` 

>I quite like having button_toggle in the config

This adds `button_toggle` too.

~~EDIT: Wanted to know if I should replace `system` w/ `self.py3.command_run`.~~ Did that.
EDIT: Wanted to know if we should replace that one `system`.

### CODE REVIEW

File: py3status/modules/dpms.py [hunk 1]
```diff
@@ -52,10 +56,11 @@ def dpms(self):
         """
         Display a colorful state of DPMS.
         """
-        self.run = system('xset -q | grep -iq "DPMS is enabled"') == 0
+        self.run = system('xset -q | grep -iq "DPMS is Enabled"') == 0
```
Comment (tobes) [inline py3status/modules/dpms.py]: `grep -i` is case insensitive.

File: py3status/modules/dpms.py [hunk 2]
```diff
@@ -7,6 +7,8 @@
 by clicking on 'DPMS' in the status bar.
 
 Configuration parameters:
+    button_toggle: mouse button to toggle DPMS (default 1)
+    cache_timeout: refresh interval for this module (default 60)
```
Comment (lasers) [inline py3status/modules/dpms.py]: Calling `xset -q` every 5 seconds? I don't want `xrandr module cause screen freeze for ~0.5 second every time it refreshes` issue for dpms.py. Still do 5? 

I see `-i` below. Just putting in what it printed out accurately.

Comment (tobes) [inline py3status/modules/dpms.py]: > Calling xset -q | grep -iq ... (two things) every 5 seconds? I don't want xrandr module cause screen freeze for ~0.5 second every time it refreshes issue for dpms.

I don't think they are related.  that is a xrandr specific issue.  how about 15?

**Comment by lasers** [comment]
>Looks good. There are other things in this module that could be improved but that is all for another day.

What is it? Anyhow, timeout_cache is updated with new number. Ready to go.

**Comment by tobes** [comment]
> What is it?

The whole `self.run` for a start.  When we click on it there is a chance (maybe small) that the value is incorrect.  It also adds unneeded complexity to the code

I'd also like to kill `system` at some point but that'd need to be done elsewhere (py3) and needs thinking about.

**Comment by lasers** [comment]
`self.run` -> `run`. Now we check dpms w/ `on_click too`.

File: py3status/modules/dpms.py [hunk 3]
```diff
@@ -52,26 +56,29 @@ def dpms(self):
...
+
+        if event['button'] == self.button_toggle:
+            if run:
+                run = False
+                self.py3.command_run("xset -dpms s off")
             else:
-                self.run = True
-                system("xset +dpms;xset s on")
+                run = True
+                self.py3.command_run("xset +dpms s on")
```
Comment (tobes) [inline py3status/modules/dpms.py]: If we are being picky then we only need to check if we are running when we toggle and the `run = True\False` lines are not needed.

**Comment by lasers** [comment]
@tobes Removed unnecessary `run` flags. Added new feature.

**Comment by tobes** [comment]
you had me till

> Added new feature.

maybe have `button_off = None` as the default.  Users may have their own onclick action defined and now you break this for them.

**Comment by lasers** [comment]
I now think 3 might be a bad default, but Idk about `None.`
What about 2? Lot of apps uses middle-mouse to close windows/tabs/etc.

**Comment by tobes** [comment]
> I now think 3 might be a bad default, but Idk about None.
What about 2? Lot of apps uses middle-mouse to close windows/tabs/etc.

__Users may have their own onclick action defined and now you break this for them.__

**Comment by lasers** [comment]
Pushed out w/ `None`!

**Comment by ultrabug** [comment]
Since you're working hard on modules usage of helpers, please get rid of the `system` call @lasers ?

**Comment by lasers** [comment]
I can't get this to work with `self.py3.command_output`.
`Exception: Command 'xset' had error xset:  unknown option |`

Any suggestions? I tried `\|` too.

**Comment by lasers** [comment]
@tobes Is there anything I can do here or should the tag go back up?

**Comment by tobes** [comment]
> should the tag go back up

It can when we've resolved @ultrabug's concern.  I'll take a look.

**Comment by tobes** [comment]
```run = 'DPMS is Enabled' in self.py3.command_output('xset -q')```

**Comment by lasers** [comment]
Thank you. @tobes. Pushed.

**Comment by lasers** [comment]
You can see how this could be a bit messy. It's `{icon}` right now, but it may feel like `{state}` would fit the purpose better here. That's why I wanted to go with generic strings to avoid this kind of simple/stupid conflict.

I looked through this again. I feel it's ready for a tag. Thank you.

File: py3status/modules/dpms.py [hunk 4]
```diff
@@ -48,30 +52,43 @@ class Meta:
...
+
     def dpms(self):
         """
         Display a colorful state of DPMS.
         """
-        self.run = system('xset -q | grep -iq "DPMS is enabled"') == 0
-        icon = self.icon_on if self.run else self.icon_off
+        run = 'DPMS is Enabled' in self.py3.command_output('xset -q')
+
+        if run:
```
Comment (tobes) [inline py3status/modules/dpms.py]: why not just `if 'DPMS is Enabled' in self.py3.command_output('xset -q')`

Comment (lasers) [inline py3status/modules/dpms.py]: Python is crazy. Yes. I like it. Less lines. How to 1s delay `xset dpms force off`? Time to get away a bit.. Avoids waking up DPMS again.

**Comment by tobes** [comment]
@lasers nice I like the button_off new feature

**Comment by ultrabug** [comment]
Thanks a lot guys !

**Pull request merged**